crates/router/src/core/unified_connector_service/utils.rs

```
@@ -0,0 +1,556 @@
+use api_models::admin::ConnectorAuthType;
+use common_enums::{enums::PaymentMethod, AttemptStatus, AuthenticationType};
+use common_utils::ext_traits::ValueExt;
+use error_stack::ResultExt;
+use hyperswitch_connectors::utils::CardData;
+use hyperswitch_domain_models::{
+    errors::api_error_response::ApiErrorResponse,
+    merchant_context::MerchantContext,
+    router_data::{ErrorResponse, RouterData},
+    router_flow_types::payments::Authorize,
+    router_request_types::PaymentsAuthorizeData,
+    router_response_types::PaymentsResponseData,
+};
+use masking::{ExposeInterface, PeekInterface};
+use rand::Rng;
+use rust_grpc_client::payments::{
+    self as payments_grpc, payment_service_client::PaymentServiceClient,
+};
+use tonic::metadata::MetadataMap;
+
+use crate::{
+    consts::UCS_ROLLOUT_PERCENT_CONFIG_PREFIX,
+    core::{
+        errors::RouterResult, payments::helpers::MerchantConnectorAccountType, utils::get_flow_name,
+    },
+    routes::SessionState,
+    types::transformers::ForeignTryFrom,
+};
+
+pub async fn should_call_unified_connector_service<F: Clone, T>(
+    state: &SessionState,
+    merchant_context: &MerchantContext,
+    router_data: &RouterData<F, T, PaymentsResponseData>,
+) -> RouterResult<Option<PaymentServiceClient<tonic::transport::Channel>>> {
+    let merchant_id = merchant_context
+        .get_merchant_account()
+        .get_id()
+        .get_string_repr();
+
+    let connector_name = router_data.connector.clone();
+
+    let payment_method = router_data.payment_method.to_string();
+    let flow_name = get_flow_name::<F>()?;
+
+    let config_key = format!(
+        "{}_{}_{}_{}_{}",
+        UCS_ROLLOUT_PERCENT_CONFIG_PREFIX, merchant_id, connector_name, payment_method, flow_name
+    );
+
+    let db = state.store.as_ref();
+
+    match db.find_config_by_key(&config_key).await {
+        Ok(rollout_config) => match rollout_config.config.parse() {
```

This functionally should be util function and not `should_call_unified_connector_service` as util function,

Check the comments raised by coderabbitai mainly,

- rollout logic
- Client creation in every call is in-efficient
- Don't discard errors

---

I have resolved it, thanks.